yrs/src/update.rs

```diff
@@ -183,35 +183,37 @@ impl Update {
                 self.blocks.clients.keys().cloned().collect();
             client_block_ref_ids.sort();
 
-            let mut current_client_id = client_block_ref_ids.pop();
-            let mut current_target =
-                current_client_id.and_then(|id| self.blocks.clients.get_mut(&id));
-            let mut stack_head = Self::next(&mut current_target);
+            let mut current_client_id = client_block_ref_ids.pop().unwrap();
+            let mut current_target = self.blocks.clients.get_mut(&current_client_id);
+            let mut stack_head = if let Some(v) = current_target.as_mut() {
+                v.pop_front()
+            } else {
+                None
+            };
 
             let mut local_sv = store.blocks.get_state_vector();
             let mut missing_sv = StateVector::default();
             let mut remaining = UpdateBlocks::default();
             let mut stack = Vec::new();
 
             while let Some(mut block) = stack_head {
-                let id = block.id();
-                if local_sv.contains(id) {
+                let id = *block.id();
+                if local_sv.contains(&id) {
                     let offset = local_sv.get(&id.client) as i32 - id.clock as i32;
                     if let Some(dep) = Self::missing(&block, &local_sv) {
                         stack.push(block);
                         // get the struct reader that has the missing struct
                         match self.blocks.clients.get_mut(&dep) {
                             Some(block_refs) if !block_refs.is_empty() => {
-                                current_target = Some(block_refs);
```

Comment:
If `current_target` was not empty in this branch, it would be dropped and remaining blocks would never be integrated.